core/node/core.go

```diff
@@ -59,15 +59,19 @@ func Dag(bs blockservice.BlockService) format.DAGService {
 }
 
 // OnlineExchange creates new LibP2P backed block exchange (BitSwap)
-func OnlineExchange(mctx helpers.MetricsCtx, lc fx.Lifecycle, host host.Host, rt routing.IpfsRouting, bs blockstore.GCBlockstore) exchange.Interface {
-	bitswapNetwork := network.NewFromIpfsHost(host, rt)
-	exch := bitswap.New(helpers.LifecycleCtx(mctx, lc), bitswapNetwork, bs)
-	lc.Append(fx.Hook{
-		OnStop: func(ctx context.Context) error {
-			return exch.Close()
-		},
-	})
-	return exch
+func OnlineExchange(provide bool) interface{} {
+	return func(mctx helpers.MetricsCtx, lc fx.Lifecycle, host host.Host, rt routing.IpfsRouting, bs blockstore.GCBlockstore) exchange.Interface {
+		bitswapNetwork := network.NewFromIpfsHost(host, rt)
+		bitswap.ProvideEnabled = provide
```

Comment:
Understood. I used the `UseHAMTSharding` experimental flag as prior art, but I will make this change. This will require a change in `go-bitswap` as that has already been merged.

Reply:
:disappointed:... Yeah, I think that's because _that_ one:

1. Is hard to feed through. It isn't used in a specific subsystem but by anything that creates a directory.
2. Was probably supposed to graduate from experimental pretty quickly.

Reply:
Cool. Thanks for pointing it out, no a big change. I'll push shortly.